sqlglot/optimizer/expand_join_constructs.py

```diff
@@ -0,0 +1,25 @@
+from sqlglot import exp
+
+
+def expand_join_constructs(expression: exp.Expression) -> exp.Expression:
```

Comment:
can we move this into qualify_tables?

this is basically a required step and i don't want to manually always have to do qualify tables, expand joins, qualify_columns.

then we won't have two do an additonal walk as well

Reply:
I'll try to merge them yeah

Reply:
Ok moved in `qualify_tables`. This required some minor changes in the scope module to deal with the "special" AST form of a "join construct", for example:

```
(TABLE this:
  (IDENTIFIER this: tbl1, quoted: False), joins:
  (JOIN this:
    (TABLE this:
      (IDENTIFIER this: tbl2, quoted: False)), on:
    (EQ this:
      (COLUMN this:
        (IDENTIFIER this: id1, quoted: False)), expression:
      (COLUMN this:
        (IDENTIFIER this: id2, quoted: False)))))
```